validator/src/main/java/com/google/daq/mqtt/mapping/MappingService.java

```diff
@@ -0,0 +1,92 @@
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MappingService {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MappingService.class);
+  private static final String SERVICE_NAME = "MappingService";
+  private SourceRepository sourceRepository;
+  private final int REQUIRED_ARGUMENT_LENGTH = 5;
+  private final String registryId;
+  private final String projectId;
+  private final String baseCloningDirectory;
+  private final String projectSpec;
+  private final String mappingFamily;
+  private final String discoveryNodeDeviceId;
+  private static final String IMPORT_BRANCH = "main";
+
+  /**
+   * Mapping Service for mapping the discovery results
+   * accepts commandlineArguments as
+   * registryId projectId baseCloningDirectory projectSpec mappingFamily discoveryNodeDeviceId
+   *
+   * registryId: e.g. ZZ-TRI-FECTA
+   * projectId: GCP project id
+   * baseCloningDirectory: local path for the repository
+   * projectSpec: eg. //mqtt/localhost
+   * mappingFamily:vendor/bacnet
+   * discoveryNodeDeviceId: discoveryNode deviceId e.g. GAT-123
+   */
+  public static void main(String[] args) {
+    MappingService mappingService = new MappingService(args);
+    mappingService.initialize();
+
+    mappingService.cloneRepo();
+    mappingService.process();
+  }
+
+  public MappingService(String[] args) {
+    if (args.length != REQUIRED_ARGUMENT_LENGTH) {
+      throw new IllegalArgumentException("Invalid arguments provided");
+    }
+
+    this.registryId = args[0];
+    this.projectId = args[1];
+    this.baseCloningDirectory = args[2];
+    this.projectSpec = args[3];
+    this.mappingFamily = args[4];
+    this.discoveryNodeDeviceId = args[5];
+  }
+
+  private void cloneRepo() {
+    if (!sourceRepository.clone(IMPORT_BRANCH)) {
+      throw new RuntimeException("Could not clone repository " + registryId);
+    }
+  }
+
+  private void initialize() {
+    String repoCloneDir = Paths.get(baseCloningDirectory, registryId).toString();
+    sourceRepository = new SourceRepository(registryId, repoCloneDir, null, projectId, null);
+
+  }
+
+  private void process() {
+    String timestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMdd.HHmmss"));
+
+    String exportBranch = String.format("proposal/%s/%s", SERVICE_NAME, timestamp);
+    if (!sourceRepository.checkoutNewBranch(exportBranch)) {
+      throw new RuntimeException("Unable to create and checkout export branch " + exportBranch);
+    }
+
+    String udmiModelPath = Paths.get(sourceRepository.getDirectory(), "udmi").toString();
+    (new Registrar()).processArgs(new ArrayList<>(List.of(udmiModelPath, projectSpec))).execute();
+    MappingAgent mappingAgent = new MappingAgent(new ArrayList<>(List.of(udmiModelPath, projectSpec)));
+    mappingAgent.setDiscoveryNodeDeviceId(this.discoveryNodeDeviceId);
+    mappingAgent.process(new ArrayList<>(List.of("map", mappingFamily)));
```

Comment:
This seems a bit odd to be "simulating" a command line when it's a direct call on the class.  I'd consider extracting a method inside of MappingAgent that this can call directly, or at the _very_ least, if that's too messy for some reason (can't think why), use a named constant for "map" so it's crystal clear what the string is doing.